### PR TITLE
Add CLI.

### PR DESCRIPTION
**Pull request by Vaibhavs10**

### CODE REVIEW

File: transcribe.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        torch_dtype=torch.float16,
+        device=f"cuda:{device_id}",
+    )
+
+    pipe.model = pipe.model.to_bettertransformer()
+
+    outputs = pipe(
+        file_name,
+        chunk_length_s=30,
+        batch_size=24,
```
Comment (pcuenca) [inline transcribe.py:34]: My GPU was using 12 GB of RAM while transcribing. It has more RAM, I assume these parameters are a good balance between memory and performance?

**Review by pcuenca — COMMENTED** [review]
Amazing! I transcribed a 90 minute audio in mere seconds!

I just wrote a few ideas for future improvement :)

File: transcribe.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import json
+
+import typer
```
[the remark this answers is not in the document]
Comment (Vaibhavs10) [reply transcribe.py:3]: Good shout! Done!

File: transcribe.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    transcript_path: str,    
+    model_name: str,
+    task: str,
+    language: str,
+):
+    pipe = pipeline(
+        task="automatic-speech-recognition",
+        model=model_name,
+        torch_dtype=torch.float16,
+        device=f"cuda:{device_id}",
```
[the remark this answers is not in the document]
Comment (Vaibhavs10) [reply transcribe.py:26]: I checked last month and it didn't work. But haven't tested on the latest OS yet

File: transcribe.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# TODO: Add a function to check hardware/ GPU and choose an appropriate model
+
+
+def main(
+    file_name: str,
+    device_id: str,
+    transcript_path: str,    
+    model_name: str,
+    task: str,
+    language: str,
```
[the remark this answers is not in the document]
Comment (Vaibhavs10) [reply transcribe.py:20]: URLs should work by default with pipeline. I'll test this in a second

File: transcribe.py (hunk 1 above)
Comment (Vaibhavs10) [reply transcribe.py:34]: Yeah! Basically beyond this point we reach diminishing returns. This is the most I could get out of a Colab T4 and I assume an average joe has that.

File: transcribe.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import json
+
+import argparse
+import torch
+from transformers import pipeline
+
+parser = argparse.ArgumentParser(description="Automatic Speech Recognition")
+parser.add_argument(
+    "--file_name",
+    required=True,
```
Comment (Vaibhavs10) [inline transcribe.py:10]: Only required variable!

File: transcribe.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    "--transcript_path",
+    required=False,
+    default="output.json",
+    type=str,
+    help="Path to save the transcription output. (default: output.json)",
+)
+parser.add_argument(
+    "--model_name",
+    required=False,
+    default="openai/whisper-large-v2",
```
Comment (Vaibhavs10) [inline transcribe.py:31]: This opens the door to try newer models as well.

Comment (Vaibhavs10) [reply transcribe.py:31]: Ofc if supported in Transformers.

File: transcribe.py (hunk 1 above)
Comment (pcuenca) [reply transcribe.py]: Yes, they can tweak it manually if they want to experiment :)

File: transcribe.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    type=str,
+    help="Name of the pretrained model/ checkpoint to perform ASR. (default: openai/whisper-large-v2)",
+)
+parser.add_argument(
+    "--task",
+    required=False,
+    default="transcribe",
+    type=str,
+    choices=["transcribe", "translate"],
+    help="Task to perform, transcribe or translate to another language. (default: transcribe)",
```
Comment (pcuenca) [inline transcribe.py:41]:
```suggestion
    help="Task to perform: transcribe or translate to another language. (default: transcribe)",
```

File: transcribe.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+)
+parser.add_argument(
+    "--language",
+    required=False,
+    type=str,
+    default="en",
+    help='Language of the input audio. (default: "en" (English))',
+)
+parser.add_argument(
+    "--batch_size",
```
Comment (pcuenca) [inline transcribe.py:51]: 🙌

**Review by pcuenca — APPROVED** [review]
Awesome! 🔥

**Pull request merged**